x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/services/analytics_service/delete_analytics.ts

```diff
@@ -39,3 +36,88 @@ export const deleteAnalytics = async (d: DataFrameAnalyticsListRow) => {
   }
   refreshAnalyticsList$.next(REFRESH_ANALYTICS_LIST_STATE.REFRESH);
 };
+
+export const deleteAnalyticsAndTargetIndex = async (
```

Comment:
please create a service in this file, something like 
```typescript
export class AnalyticsService {
  constructor(toastNotifications: ToastsStart) {}
  
  deleteAnalytics() {...}
  deleteAnalyticsAndTargetIndex() {...}
}
```
and when you instantiace this service in your component, provide `toastNotifications` form kibana context.

`checkUserCanDeleteIndex` function should belong to another service I guess, because it's more generic and relevant not only to DFA.

Reply:
it also depends on `ml.dataFrameAnalytics`, so it should be injected as well

Reply:
I'll save this for a follow up PR when we refactor for dependency cache in the future.